litgpt/utils.py

```diff
@@ -516,3 +516,10 @@ def instantiate_torch_optimizer(optimizer, model_parameters, **kwargs):
         optimizer["init_args"].update(kwargs)
         optimizer = instantiate_class(model_parameters, optimizer)
     return optimizer
+
+
+def extend_checkpoint_dir(checkpoint_dir: Path) -> Path:
+    new_checkpoint_dir = Path(checkpoint_dir)  # duplicate Path object to not overwrite the original
```

Comment:
To my knowledge, the case in the comment cant occur, and `"checkpoints" / new_checkpoint_dir` returns a new path object

Reply:
You are right, just double-checked:

```python
from pathlib import Path


path_1 = Path("abc")
path_2 = path_1
path_3 = "checkpoints" / path_1

print(path_1 is path_2)
print(path_2 is path_3)

# Prints
# True
# False

```